tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+        log.info(f"Added {service_obj.daemon_type} service successfully")
+
+    with parallel() as p:
+        p.spawn(add_service)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def remove_osd_service_with_cache_test(
+    rados_obj, ceph_cluster, pwl, config, cache_client, image_spec
+):
+    """
+    Test to remove OSD service along with verifying the persistent write cache.
+
+    Args:
+        rados_obj: RadosOrchestrator object for interacting with the Ceph Orchestrator.
+        ceph_cluster: Ceph cluster object.
+        pwl: PersistentWriteAheadLog instance.
+        config: Test configuration.
+        cache_client: Client node for cache testing.
+        image_spec: Image specification for testing.
+    """
+    log.info("Test to remove OSD service along with pwl cache test")
+
+    pool_name = config["rep_pool_config"]["pool"]
+    pg_set = rados_obj.get_pg_acting_set(pool_name=pool_name)
+    target_osd = pg_set[0]
+    host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=target_osd)
+
+    with parallel() as p:
+        p.spawn(utils.osd_remove, ceph_cluster, target_osd)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+    return host
+
+
+def add_osd_service_with_cache_test(
+    ceph_cluster, host, pwl, config, cache_client, image_spec
+):
+    """
+    Test to add OSD service along with verifying the persistent write cache.
+
+    Args:
+        ceph_cluster: Ceph cluster object.
+        host (str): The name of the host.
+        pwl: PersistentWriteAheadLog instance.
+        config: Test configuration.
+        cache_client: Client node for cache testing.
+        image_spec: Image specification for testing.
+    """
+    log.info("Test to add osd service along with pwl cache test")
+    orch_obj = Orch(cluster=ceph_cluster, **config)
+    osd_obj = OSD(cluster=ceph_cluster, **config)
+    ceph_admin = CephAdmin(cluster=ceph_cluster, **config)
+
+    with parallel() as p:
+        p.spawn(
+            add_remove_osd,
+            command="add",
+            osd_nodes=host.hostname,
+            ceph_nodes=ceph_cluster,
+            orch_obj=orch_obj,
+            osd_obj=osd_obj,
+            ceph_admin=ceph_admin,
+        )
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def initiate_osd_scrub_test_with_cache(cache_client, pwl, config, image_spec):
+    """
+    Test to initiate OSD scrub along with verifying the persistent write cache.
+
+    Args:
+        cache_client: Client node for cache testing.
+        pwl: PersistentWriteAheadLog instance.
+        config: Test configuration.
+        image_spec: Image specification for testing.
+    """
+    log.info("Test to initiate osd scrub along with pwl cache test")
+
+    def initiate_osd_scrub():
+        # Initiate OSD scrub
+        cmd = "ceph osd scrub all"
+        cache_client.exec_command(cmd=cmd, sudo=True)
+
+    with parallel() as p:
+        p.spawn(initiate_osd_scrub)
```

Comment:
This implementation still doesn't catch any return value and throw exception right? As we discussed.